src/nomad_parser_h5md/parsers/h5md_parser.py

```diff
@@ -29,90 +33,70 @@ def get_value(self, name: str, dct: Dict[str, Any]) -> Any:
             return None
 
-        return self.get_value(kwargs.get('key'), source)
+        value = self.get_value(kwargs.get('key'), source)
 
-    def get_system_hierarchy(
-        self,
-        particlesgroup: {},
-    ) -> List[Dict[str, Any]]:
-        data = []
-        for key, dct in particlesgroup.items():
-            data.append(dct)
-            path_particlesgroup_key = f'{path_particlesgroup}.{key}'
+        enum_spec = kwargs.get('enum_spec', None)
+        if enum_spec == 'upper':
+            return value.upper() if isinstance(value, str) else value
+        elif enum_spec == 'lower':
+            return value.lower() if isinstance(value, str) else value
 
-            particles_group = {
-                group_key: h5md_sec_particlesgroup.get(
-                    f'{path_particlesgroup_key}.{group_key}'
-                )
-                for group_key in h5md_sec_particlesgroup[key].keys()
-            }
+        return value
 
-            particles_group = {
-                group_key: self.data.get(f'{path_particlesgroup_key}.{group_key}')
-                for group_key in h5md_sec_particlesgroup[key].keys()
-            }
-            data['branch_label'] = particles_group.pop('label', None)
-            data['atom_indices'] = particles_group.pop('indices', None)
-            # TODO remove the deprecated below from the test file
-            data['type'] = particles_group.pop('type', None)  # ? deprecate?
-            data['is_molecule'] = particles_group.pop(
-                'is_molecule', None
-            )  # ? deprecate?
-            particles_group.pop('formula', None)  # covered in normalization now
-            # write all the standard quantities to the archive
-            particles_subgroup = particles_group.pop('particles_group', None)
-
-            # set the remaining attributes
-            data['custom_system_attributes'] = []
-            for particles_group_key in particles_group.keys():
-                val = particles_group.get(particles_group_key)
-                units = val.units if hasattr(val, 'units') else None
-                val = val.magnitude if units is not None else val
-                data['custom_system_attributes'].append(
-                    {'name': particles_group_key, 'value': val, 'unit': units}
-                )
+    def get_sub_systems(self, source: dict[str, Any], **kwargs) -> list[dict[str, Any]]:
+        step = source.get('step', None)
+        if step is not None:
+            if step != 0:  # TODO extend to time-dependent bond lists and topologies
+                return []
+            source = self.get_source(self.data, kwargs['path'])
 
-            # get the next branch level
-            if particles_subgroup:
-                self.get_system_hierarchy(
-                    particles_subgroup,
-                    f'{path_particlesgroup_key}.particles_group',
-                )
+        particles_group = source.get('particles_group', None)
```

Comment:
maybe set default here to `{}` so you get rid of the l66-67 and the `if` condition in l71